_delphi_utils_python/delphi_utils/runner.py

```diff
@@ -66,13 +67,16 @@ def run_indicator_pipeline(indicator_fn:  Callable[[Params], None],
     validator = validator_fn(params)
     archiver = archiver_fn(params)
 
-    t = threading.Thread(target=flash_fn, args=[params])
-    t.start()
-    t.join(timer)
-    if t.is_alive():
-        t.join()
-
-
+    t1 = multiprocessing.Process(target=flash_fn, args=[params])
+    t1.start()
+    start = time.time()
+    while time.time()-start < timer:
+        if not t1.is_alive():
+            break
+        time.sleep(10)
+    if t1.is_alive():
```

Comment:
suggestion: this is an ideal candidate for `while/else`

```suggestion
    else:
```

The `else` clause executes only when `while` exits without breaking; https://stackoverflow.com/a/3295949/551433